notification.c

```diff
@@ -135,8 +137,109 @@ static size_t escape_markup(const char *s, char *buf) {
 	return len;
 }
 
-size_t format_notification(struct mako_notification *notif, const char *format,
-		char *buf) {
+char *format_state_text(char variable, void *data) {
+	struct mako_state *state = (struct mako_state *)data;
+	char *value = NULL;
+	switch (variable) {
+	case 'h':;
+		int hidden = wl_list_length(&state->notifications) - state->config.max_visible;
+		size_t hidden_ln = snprintf(NULL, 0, "%d", hidden);
+		value = malloc(hidden_ln + 1);
+		if(!value) {
+			break;
+		}
+		snprintf(value, hidden_ln + 1, "%d", hidden);
+		return value;
+	case 't':;
+		int count = wl_list_length(&state->notifications);
+		size_t total_ln = snprintf(NULL, 0, "%d", count);
```

Comment:
Was thinking about making this "int to char array" into a function since I've already used it twice.

Reply:
+1 for implementing `asprintf`

This function is not part of the standard C library - it's a GNU extension. We don't use GNU extensions because they're not portable and not part of the standard.

Reply:
Should this function be added to notification.c? Or would it be better to create a tool/utility header so so we include elsewhere if needed?

Reply:
Let's just add it here as a static function for now, and export it if we need it elsewhere.